tools/ClangTidyHash.sh

```diff
@@ -29,9 +29,35 @@ done
 
 # Go to build directory and then run clang-tidy, writing output to file
 cd ${BUILD_DIR}
+CLANG_TIDY_OUTPUT="./.clang_tidy_check_output"
+rm -f ${CLANG_TIDY_OUTPUT}
+
+printf "\nRunning clang-tidy on:\n"
+echo $MODIFIED_FILES
 
 for FILENAME in ${MODIFIED_FILES[@]}
 do
-    printf "\n\nRunning clang-tidy on file $FILENAME\n"
-    make clang-tidy FILE=$SOURCE_DIR/${FILENAME}
+    printf "\n\nRunning clang-tidy on file $FILENAME\n" >> ${CLANG_TIDY_OUTPUT}
+    make clang-tidy FILE=$SOURCE_DIR/${FILENAME} >> ${CLANG_TIDY_OUTPUT} 2>&1
```

Comment:
Is it possible to use `--warnings-as-errors` to have this return a non-zero exit code when a warning is generated? Then you don't need to generate a file. If not, you should save the output to a variable rather than writing it to disk. Once this is figured out, the rest of the code will need to be adjusted accordingly.

Reply:
Yes the flag works very nicely, thanks for the suggestion. I pushed a fixup, and also a commit that breaks clang-tidy so you can see how it interacts with the CI. I'll drop the commit when squashing the fixup.